cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetProperties.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import co.cask.cdap.api.annotation.Beta;
+import com.google.common.base.Joiner;
+
+import java.util.Map;
+
+/**
+ * Helper to build properties for files datasets.
+ */
+@Beta
+public class PartitionedFileSetProperties extends FileSetProperties {
+
+  /**
+   * The property name for the list of partitioning field names.
+   */
+  public static final String PARTITIONING_FIELDS = "partitioning.fields.";
+
+  /**
+   * The prefix for fields of the partitioning.
+   */
+  public static final String PARTITIONING_FIELD_PREFIX = "partitioning.field.";
+
+  /**
+   * Read the partitioning for a PartitionedFileSet from its properties.
+   *
+   * @param properties the dataset properties
+   */
+  public static Partitioning getPartitioning(Map<String, String> properties) {
```

Comment:
Looks like this method can return `null` for `Partitioning`. It's better to annotate this method with `@Nullable` especially it's cdap-api.